### DISCUSSION TITLE
Multibindings support

### DISCUSSION THREAD

**Discussion by badoualy** [sczerwinski/android-hilt#140] [Ideas]
Not sure if this is something that should be done, maybe it's going too far.

Support for multibindings could be interesting
Equivalent of `IntoMap` or `IntoSet`

eg:

```
@BoundIntoMap(component = ..., supertype = ...)
class Foo @Inject constructor(): Bar
```

**Comment by sczerwinski** [comment]
This would be a reasonable feature.

While adding support for `@IntoSet` and `@ElementsIntoSet` would be pretty straightforward, I'm not sure how to provide keys for `@IntoMap`. There are: `@StringKey`, `@ClassKey` and any custom annotation annotated with `@MapKey` (I can see reasonable solution for the first two).

Also, currently, there are 6 annotations:
* `@Bound`
* `@BoundTo`
* `@FactoryMethod`
* `@TestBound`
* `@TestBoundTo`
* `@TestFactoryMethod`
so instead of creating additional 3+ variants of each, I'd rather define parameter(s) in existing ones.

I'll need some time to think it though. As always, any suggestions are welcome.

**Comment by sczerwinski** [comment]
I still haven't figured out a nice way to do it. But I'm open to any suggestions/pull requests.

Moving the issue to discussions.
